tests/testthat/test-cache.R

```diff
@@ -0,0 +1,55 @@
+context("Cache")
+
+withTestContext({
+  # https://github.com/rstudio/packrat/issues/345
+  test_that("package installation when configured with a a cache uses the cache", {
+    skip_on_cran()
+
+    projRoot <- cloneTestProject("healthy")
+    libRoot <- file.path(projRoot,"packrat","lib")
+    srcRoot <- file.path(projRoot,"packtar","src")
+    theCache <- tempdir()
+
+    packrat::init(projRoot, options = list(local.repos = "packages"))
+
+    rv <- R.Version()
+    package_dir <- file.path(projRoot,
+                             "packrat",
+                             "lib",
+                             rv$platform,
+                             getRversion(),
+                             "oatmeal")
+
+    # After the init and before we configure a cache, the package is a
+    # non-symlink. dir.exists only tells us if the package is installed, as it
+    # reads through symlinks.
+    expect_true(dir.exists(package_dir),package_dir)
+    expect_false(packrat:::is.symlink(package_dir), package_dir)
```

Comment:
nit: I believe `testthat` sources these files with all unexported package objects made available as though they were exported (e.g. you can just use `is.symlink()` without the `packrat` prefix).